webknossos-datastore/app/com/scalableminds/webknossos/datastore/datareaders/ChunkReader.scala

```diff
@@ -44,104 +44,107 @@ class ChunkReader(val header: DatasetHeader, val store: FileSystemStore, val typ
 abstract class TypedChunkReader {
   val header: DatasetHeader
 
-  var chunkShape: Array[Int] = header.chunkShapeOrdered
   def ma2DataType: MADataType
-  def read(bytes: Option[Array[Byte]]): Future[MultiArray]
+  def read(bytes: Option[Array[Byte]], chunkShape: Array[Int]): Future[MultiArray]
 
-  def createFilled(dataType: MADataType): MultiArray =
+  def createFilled(dataType: MADataType, chunkShape: Array[Int]): MultiArray =
     MultiArrayUtils.createFilledArray(dataType, chunkShape, header.fillValueNumber)
+
+  // Chunk shape in header is in C-Order (XYZ), but data may be in F-Order (ZYX), so the chunk shape
+  // associated with the array needs to be adjusted for non-isotropic chunk sizes.
+  def chunkDataShape(chunkShape: Array[Int]) = if (header.order == ArrayOrder.F) chunkShape.reverse else chunkShape
```

Comment:
@jstriebel proposed this name to display that the chunk will be returned in the shape of the data, I'm fine with it either way

Reply:
@jstriebel I hope you are ok with my suggestion as well? When reading the code, I did not understand at first what chunkDataShape is (and why it would differ from chunkShape). The comment clears is up ultimately, but I still like my name suggestion better :sweat_smile: